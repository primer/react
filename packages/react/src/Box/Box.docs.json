{
  "id": "box",
  "name": "Box",
  "status": "beta",
  "a11yReviewed": false,
  "stories": [],
  "importPath": "@primer/react",
  "props": [
    {
      "name": "ref",
      "type": "React.RefObject<HTMLDivElement>"
    },
    {
      "name": "as",
      "type": "React.ElementType",
      "defaultValue": "\"div\""
    },
    {
      "name": "sx",
      "type": "SystemStyleObject"
    }
  ],
  "subcomponents": []
}