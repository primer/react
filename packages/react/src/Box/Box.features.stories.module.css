.Border {
  border: var(--borderWidth-thin) solid var(--borderColor-default);
  padding: var(--base-size-16);
}

.BorderBottom {
  border-bottom: var(--borderWidth-thin) solid var(--borderColor-default);
  padding: var(--base-size-16);
}

.FlexContainer {
  display: flex;
}

.FlexItem {
  padding: var(--base-size-16);
  border: var(--borderWidth-thin) solid var(--borderColor-default);
}

.FlexItemGrow {
  padding: var(--base-size-16);
  border: var(--borderWidth-thin) solid var(--borderColor-default);
  flex-grow: 1;
}

.GridContainer {
  display: grid;
  grid-template-columns: 1fr 1fr;
  gap: var(--base-size-16);
}

.GridItem {
  padding: var(--base-size-16);
  border: var(--borderWidth-thin) solid var(--borderColor-default);
}
