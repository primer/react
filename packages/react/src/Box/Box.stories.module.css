.PlaygroundBox {
  border-width: var(--borderWidth-thin);
  border-style: solid;
  border-color: var(--borderColor-default);
  padding: var(--base-size-12);
}
