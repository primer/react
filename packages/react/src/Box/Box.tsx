import styled from 'styled-components'
import type {
  BackgroundProps,
  BorderProps,
  ColorProps,
  FlexboxProps,
  GridProps,
  LayoutProps,
  PositionProps,
  ShadowProps,
  SpaceProps,
  TypographyProps,
} from 'styled-system'
import {background, border, color, flexbox, grid, layout, position, shadow, space, typography} from 'styled-system'
import type {SxProp} from '../sx'
import sx from '../sx'
import type {ComponentProps} from '../utils/types'

type StyledBoxProps = SpaceProps &
  ColorProps &
  TypographyProps &
  LayoutProps &
  FlexboxProps &
  GridProps &
  BackgroundProps &
  BorderProps &
  PositionProps &
  ShadowProps &
  SxProp

const Box = styled.div<StyledBoxProps>(
  space,
  color,
  typography,
  layout,
  flexbox,
  grid,
  background,
  border,
  position,
  shadow,
  sx,
)

export type BoxProps = ComponentProps<typeof Box>
export default Box
