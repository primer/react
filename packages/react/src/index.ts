'use client'

export {default as theme} from './theme'
export {get as themeGet} from './constants'
export {default as BaseStyles} from './BaseStyles'
export type {BaseStylesProps} from './BaseStyles'
export {default as ThemeProvider, useTheme, useColorSchemeVar} from './ThemeProvider'
export type {ThemeProviderProps} from './ThemeProvider'

// Layout
export {default as Box} from './Box'
export type {BoxProps} from './Box'
export * from './Button'
export {PageLayout} from './PageLayout'
export type {
  PageLayoutProps,
  PageLayoutHeaderProps,
  PageLayoutContentProps,
  PageLayoutPaneProps,
  PageLayoutFooterProps,
} from './PageLayout'
export {SplitPageLayout} from './SplitPageLayout'
export type {
  SplitPageLayoutProps,
  SplitPageLayoutHeaderProps,
  SplitPageLayoutContentProps,
  SplitPageLayoutPaneProps,
  SplitPageLayoutFooterProps,
} from './SplitPageLayout'

// Hooks
export {default as useDetails} from './hooks/useDetails'
export {default as useSafeTimeout} from './hooks/useSafeTimeout'
export {useOnOutsideClick} from './hooks/useOnOutsideClick'
export type {TouchOrMouseEvent} from './hooks/useOnOutsideClick'
export {useOpenAndCloseFocus} from './hooks/useOpenAndCloseFocus'
export {useOnEscapePress} from './hooks/useOnEscapePress'
export {useOverlay} from './hooks/useOverlay'
export {useConfirm} from './ConfirmationDialog/ConfirmationDialog'
export {useFocusTrap} from './hooks/useFocusTrap'
export type {FocusTrapHookSettings} from './hooks/useFocusTrap'
export {useFocusZone} from './hooks/useFocusZone'
export type {FocusZoneHookSettings} from './hooks/useFocusZone'
export {useRefObjectAsForwardedRef} from './hooks/useRefObjectAsForwardedRef'
export {useResizeObserver} from './hooks/useResizeObserver'
export {useResponsiveValue} from './hooks/useResponsiveValue'

// Utils
export {createComponent} from './utils/create-component'

// Components
export {default as Radio} from './Radio'
export type {RadioProps} from './Radio'
export {ActionList} from './ActionList'
export type {
  ActionListProps,
  ActionListGroupProps,
  ActionListItemProps,
  ActionListLinkItemProps,
  ActionListDividerProps,
  ActionListDescriptionProps,
  ActionListLeadingVisualProps,
  ActionListTrailingVisualProps,
} from './ActionList'
export {ActionMenu} from './ActionMenu'
export type {ActionMenuProps, ActionMenuAnchorProps, ActionMenuButtonProps} from './ActionMenu'
export {AnchoredOverlay} from './AnchoredOverlay'
export type {AnchoredOverlayProps} from './AnchoredOverlay'
export {default as Autocomplete} from './Autocomplete'
export type {AutocompleteMenuProps, AutocompleteInputProps, AutocompleteOverlayProps} from './Autocomplete'
export {default as Avatar} from './Avatar'
export type {AvatarProps} from './Avatar'
export {default as AvatarPair} from './AvatarPair'
export type {AvatarPairProps} from './AvatarPair'
export {default as AvatarStack} from './AvatarStack'
export type {AvatarStackProps} from './AvatarStack'
export {default as BranchName} from './BranchName'
export type {BranchNameProps} from './BranchName'
export {default as Breadcrumbs, Breadcrumb} from './Breadcrumbs'
export type {BreadcrumbsProps, BreadcrumbsItemProps, BreadcrumbProps, BreadcrumbItemProps} from './Breadcrumbs'
export {default as ButtonGroup} from './ButtonGroup'
export type {ButtonGroupProps} from './ButtonGroup'
export {default as Caret} from './Caret'
export type {CircleBadgeProps, CircleBadgeIconProps} from './CircleBadge'
export {default as CircleOcticon} from './CircleOcticon'
export type {CircleOcticonProps} from './CircleOcticon'
export {default as CheckboxGroup} from './CheckboxGroup'
export type {CaretProps} from './Caret'
export {default as CircleBadge} from './CircleBadge'
export {default as CounterLabel} from './CounterLabel'
export type {CounterLabelProps} from './CounterLabel'
export {default as Details} from './Details'
export type {DetailsProps} from './Details'
export {default as Dialog} from './Dialog'
export type {DialogProps, DialogHeaderProps} from './Dialog'
export type {ConfirmationDialogProps} from './ConfirmationDialog/ConfirmationDialog'
export {ConfirmationDialog} from './ConfirmationDialog/ConfirmationDialog'
export {default as Flash} from './Flash'
export type {FlashProps} from './Flash'
export {default as FormControl} from './FormControl'
export {useFormControlForwardedProps} from './FormControl'
export {default as Header} from './Header'
export type {HeaderProps, HeaderItemProps, HeaderLinkProps} from './Header'
export {default as Heading} from './Heading'
export type {HeadingProps} from './Heading'
export {default as Label} from './Label'
export type {LabelProps} from './Label'
export {default as LabelGroup} from './LabelGroup'
export type {LabelGroupProps} from './LabelGroup'
export {default as Link} from './Link'
export type {LinkProps} from './Link'
export {NavList} from './NavList'
export type {
  NavListProps,
  NavListItemProps,
  NavListSubNavProps,
  NavListGroupProps,
  NavListLeadingVisualProps,
  NavListTrailingVisualProps,
  NavListDividerProps,
} from './NavList'
export {default as Octicon} from './Octicon'
export type {OcticonProps} from './Octicon'
export {default as Overlay} from './Overlay'
export type {OverlayProps} from './Overlay'
export {default as Pagehead} from './Pagehead'
export type {PageheadProps} from './Pagehead'
export {default as Pagination} from './Pagination'
export type {PaginationProps} from './Pagination'
export {default as PointerBox} from './PointerBox'
export type {PointerBoxProps} from './PointerBox'
export {default as Popover} from './Popover'
export type {PopoverProps, PopoverContentProps} from './Popover'
export {default as Portal, registerPortalRoot} from './Portal'
export type {PortalProps} from './Portal'
export {ProgressBar} from './ProgressBar'
export type {ProgressBarProps} from './ProgressBar'
export {default as RadioGroup} from './RadioGroup'
export type {RelativeTimeProps} from './RelativeTime'
export {default as RelativeTime} from './RelativeTime'
export {SegmentedControl} from './SegmentedControl'
export {default as Select} from './Select'
export type {SelectProps} from './Select'
export {SelectPanel} from './SelectPanel'
export type {SelectPanelProps} from './SelectPanel'
export {default as SideNav} from './SideNav'
export type {SideNavProps, SideNavLinkProps} from './SideNav'
export {default as Spinner} from './Spinner'
export type {SpinnerProps} from './Spinner'
export {default as StateLabel} from './StateLabel'
export type {StateLabelProps} from './StateLabel'
export {default as SubNav} from './SubNav'
export type {SubNavProps, SubNavLinkProps, SubNavLinksProps} from './SubNav'
export {default as ToggleSwitch} from './ToggleSwitch'
export {default as TabNav} from './TabNav'
export type {TabNavProps, TabNavLinkProps} from './TabNav'
export {default as TextInput} from './TextInput'
export type {TextInputProps} from './TextInput'
export {default as TextInputWithTokens} from './TextInputWithTokens'
export type {TextInputWithTokensProps} from './TextInputWithTokens'
export {default as Text} from './Text'
export type {TextProps} from './Text'
export {default as Timeline} from './Timeline'
export type {
  TimelineProps,
  TimelineBadgeProps,
  TimelineBodyProps,
  TimelineBreakProps,
  TimelineItemsProps,
} from './Timeline'
export {default as Token, IssueLabelToken, AvatarToken} from './Token'
export type {TokenProps} from './Token'
export {default as Tooltip} from './Tooltip/Tooltip'
export type {TooltipProps} from './Tooltip/Tooltip'
export {default as Truncate} from './Truncate'
export type {TruncateProps} from './Truncate'

export {default as Checkbox} from './Checkbox'
export type {CheckboxProps} from './Checkbox'

export {default as Textarea} from './Textarea'
export type {TextareaProps} from './Textarea'

export {TreeView} from './TreeView'
export type {
  TreeViewProps,
  TreeViewItemProps,
  TreeViewSubTreeProps,
  TreeViewVisualProps,
  TreeViewErrorDialogProps,
} from './TreeView'

export {UnderlineNav} from './UnderlineNav'
export type {UnderlineNavProps, UnderlineNavItemProps} from './UnderlineNav'

export {ActionBar} from './ActionBar'
export type {ActionBarProps} from './ActionBar'

// eslint-disable-next-line no-restricted-imports
export {SSRProvider, useSSRSafeId} from './utils/ssr'
export {default as sx, merge} from './sx'
export type {SxProp} from './sx'
