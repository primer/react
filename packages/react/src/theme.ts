import type {KeyPaths} from './utils/types/KeyPaths'
import {fontStack} from './utils/theme'
import {colorSchemes} from './legacy-theme/ts/color-schemes'

const animation = {
  easeOutCubic: 'cubic-bezier(0.33, 1, 0.68, 1)',
}

const breakpoints = ['544px', '768px', '1012px', '1280px']

const fonts = {
  normal: fontStack([
    '-apple-system',
    'BlinkMacSystemFont',
    'Segoe UI',
    'Noto Sans',
    'Helvetica',
    'Arial',
    'sans-serif',
    'Apple Color Emoji',
    'Segoe UI Emoji',
  ]),
  mono: `var(--fontStack-monospace, ${fontStack([
    'SFMono-Regular',
    'Consolas',
    'Liberation Mono',
    'Menlo',
    'Courier',
    'monospace',
  ])})`,
}

const lineHeights = {
  condensedUltra: 1,
  condensed: 1.25,
  default: 1.5,
}

const fontWeights = {
  light: 300,
  normal: 400,
  semibold: 500,
  bold: 600,
}

const borderWidths = [0, '1px']

const radii = ['0', '3px', '6px', '100px']

const sizes = {
  small: '544px',
  medium: '768px',
  large: '1012px',
  xlarge: '1280px',
}

const fontSizes = ['12px', '14px', '16px', '20px', '24px', '32px', '40px', '48px', '56px']

const space = ['0', '4px', '8px', '16px', '24px', '32px', '40px', '48px', '64px', '80px', '96px', '112px', '128px']

const theme = {
  animation,
  borderWidths,
  breakpoints,
  fonts,
  fontSizes,
  fontWeights,
  lineHeights,
  radii,
  sizes,
  space,
  colorSchemes,
}

export default theme

// NOTE: for now, ThemeColors and ThemeShadows are handcrafted types. It would be nice if these
// were exports from primitives (or a different shape but derived from those exports).

type ThemeColors = {
  fg: {
    default: string
    muted: string
    subtle: string
    onEmphasis: string
  }
  canvas: {
    default: string
    overlay: string
    inset: string
    subtle: string
  }
  border: {
    default: string
    muted: string
    subtle: string
  }

  // Roles
  neutral: {
    emphasisPlus: string
    emphasis: string
    muted: string
    subtle: string
  }
  accent: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  success: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  attention: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  severe: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  danger: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  open: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  closed: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  done: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
  sponsors: {
    fg: string
    emphasis: string
    muted: string
    subtle: string
  }
}

type ThemeShadows = {
  shadow: {
    small: string
    medium: string
    large: string
    extraLarge: string
  }
}

export type ThemeColorPaths = KeyPaths<ThemeColors>
export type ThemeShadowPaths = KeyPaths<ThemeShadows>
