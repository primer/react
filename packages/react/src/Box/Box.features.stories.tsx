import React from 'react'
import type {Meta} from '@storybook/react'
import Box from './Box'

export default {
  title: 'Components/Box/Features',
  component: Box,
} as Meta<typeof Box>

export const Border = () => (
  <Box sx={{borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default', p: 3}}>Box with border</Box>
)

export const BorderBottom = () => (
  <Box sx={{borderBottomWidth: 1, borderBottomStyle: 'solid', borderColor: 'border.default', p: 3}}>
    Box with bottom border
  </Box>
)

export const Flexbox = () => (
  <Box sx={{display: 'flex'}}>
    <Box sx={{p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>1</Box>
    <Box sx={{flexGrow: 1, p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>2</Box>
    <Box sx={{p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>3</Box>
  </Box>
)

export const Grid = () => (
  <Box sx={{display: 'grid', gridTemplateColumns: '1fr 1fr', gap: 3}}>
    <Box sx={{p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>1</Box>
    <Box sx={{p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>2</Box>
    <Box sx={{p: 3, borderWidth: 1, borderStyle: 'solid', borderColor: 'border.default'}}>3</Box>
  </Box>
)
