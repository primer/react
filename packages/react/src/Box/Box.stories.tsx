import React from 'react'
import type {Meta, StoryFn} from '@storybook/react'
import Box from './Box'

export default {
  title: 'Components/Box',
  component: Box,
} as Meta<typeof Box>

export const Default = () => <Box>Default Box</Box>

export const Playground: StoryFn<typeof Box> = args => <Box {...args}>Playground</Box>

Playground.args = {
  as: 'div',
  sx: {
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: 'border.default',
    p: 3,
  },
}

Playground.argTypes = {
  forwardedAs: {
    controls: false,
    table: {
      disable: true,
    },
  },
  ref: {
    controls: false,
    table: {
      disable: true,
    },
  },
  theme: {
    controls: false,
    table: {
      disable: true,
    },
  },
}
