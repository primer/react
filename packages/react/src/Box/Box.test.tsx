import {render, screen} from '@testing-library/react'
import {describe, expect, it} from 'vitest'
import Box from '../Box'
import theme from '../theme'

const breakpoints = [
  null,
  'screen and (min-width: 544px)',
  'screen and (min-width: 768px)',
  'screen and (min-width: 1012px)',
] as const

describe('Box', () => {
  it('should support `className` on the outermost element', () => {
    const Element = () => <Box className={'test-class-name'} />
    expect(render(<Element />).container.firstChild).toHaveClass('test-class-name')
  })

  it('should support margin', () => {
    render(<Box data-testid="box" m={1} theme={theme} />)
    const box = screen.getByTestId('box')
    expect(getCSSRuleForClass(getClassName(box.className)).style.margin).toBe(theme.space[1])
  })

  it('should support responsive margin', () => {
    const m = [0, 1, 2, 3]

    render(<Box data-testid="box" m={[0, 1, 2, 3]} theme={theme} />)

    const box = screen.getByTestId('box')
    const className = getClassName(box.className)

    for (let i = 0; i < m.length; i++) {
      let themeValue = theme.space[m[i]]
      if (themeValue === '0') {
        themeValue = '0px'
      }

      if (i === 0) {
        const cssRule = getCSSRuleForClass(className)
        expect(cssRule.style.margin).toBe(themeValue)
      } else {
        const cssRule = getCSSRuleForClass(className, breakpoints[i]!)
        expect(cssRule.style.margin).toBe(themeValue)
      }
    }
  })

  it('should support padding', () => {
    render(<Box data-testid="box" p={1} theme={theme} />)
    const box = screen.getByTestId('box')
    expect(getCSSRuleForClass(getClassName(box.className)).style.padding).toBe(theme.space[1])
  })

  it('should support responsive padding', () => {
    const p = [0, 1, 2, 3]

    render(<Box data-testid="box" p={[0, 1, 2, 3]} theme={theme} />)

    const box = screen.getByTestId('box')
    const className = getClassName(box.className)

    for (let i = 0; i < p.length; i++) {
      let themeValue = theme.space[p[i]]
      if (themeValue === '0') {
        themeValue = '0px'
      }

      if (i === 0) {
        const cssRule = getCSSRuleForClass(className)
        expect(cssRule.style.padding).toBe(themeValue)
      } else {
        const cssRule = getCSSRuleForClass(className, breakpoints[i]!)
        expect(cssRule.style.padding).toBe(themeValue)
      }
    }
  })

  it('should support display', () => {
    render(<Box data-testid="box" display="inline" theme={theme} />)
    const box = screen.getByTestId('box')
    expect(getCSSRuleForClass(getClassName(box.className)).style.display).toBe('inline')
  })

  it('should support responsive display', () => {
    const display = ['none', 'inline', 'inline-block', 'block']
    render(<Box data-testid="box" display={display} theme={theme} />)
    const box = screen.getByTestId('box')
    const className = getClassName(box.className)
    for (let i = 0; i < display.length; i++) {
      if (i === 0) {
        const cssRule = getCSSRuleForClass(className)
        expect(cssRule.style.display).toBe(display[i])
      } else {
        const cssRule = getCSSRuleForClass(className, breakpoints[i]!)
        expect(cssRule.style.display).toBe(display[i])
      }
    }
  })
})

function getClassName(className: string): string {
  // styled-components with `Box` seems to generate two classes: `sx-<hash>` and
  // `<hash>`. We are looking for the second class which seems to be the one the
  // styles are applied to in CSS.
  const classes = className.split(' ')
  return classes[1]
}

function getCSSRuleForClass(className: string, mediaText?: string): CSSStyleRule {
  for (const sheet of document.styleSheets) {
    for (const rule of sheet.cssRules) {
      if (mediaText) {
        if (rule instanceof CSSMediaRule && rule.media.mediaText === mediaText) {
          for (const innerRule of rule.cssRules) {
            if (innerRule instanceof CSSStyleRule && innerRule.selectorText === `.${className}`) {
              return innerRule
            }
          }
        }
      } else if (rule instanceof CSSStyleRule) {
        if (rule.selectorText === `.${className}`) {
          return rule
        }
      }
    }
  }

  throw new Error(`CSS rule for class "${className}" not found`)
}
