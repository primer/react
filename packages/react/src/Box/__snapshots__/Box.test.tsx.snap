// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders margin 1`] = `
.c0 {
  margin: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box renders margin 2`] = `
.c0 {
  margin: 0;
}

@media screen and (min-width:544px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    margin: 8px;
  }
}

@media screen and (min-width:1012px) {
  .c0 {
    margin: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders margin 3`] = `
.c0 {
  margin: 4px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {

}

@media screen and (min-width:544px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    margin: 4px;
  }
}

@media screen and (min-width:1012px) {
  .c0 {
    margin: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 1`] = `
.c0 {
  padding: 4px;
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 2`] = `
.c0 {
  padding: 0;
}

@media screen and (min-width:544px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding: 8px;
  }
}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box renders padding 3`] = `
.c0 {
  padding: 4px;
}

@media screen and (min-width:544px) {

}

@media screen and (min-width:768px) {

}

@media screen and (min-width:1012px) {

}

@media screen and (min-width:544px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    padding: 4px;
  }
}

@media screen and (min-width:1012px) {
  .c0 {
    padding: 16px;
  }
}

<div
  className="c0"
/>
`;

exports[`Box respects display 1`] = `
.c0 {
  display: inline;
}

<div
  className="c0"
  display="inline"
/>
`;

exports[`Box respects display 2`] = `
.c0 {
  display: inline-block;
}

<div
  className="c0"
  display="inline-block"
/>
`;

exports[`Box respects display 3`] = `
.c0 {
  display: none;
}

<div
  className="c0"
  display="none"
/>
`;

exports[`Box respects display 4`] = `
.c0 {
  display: none;
}

@media screen and (min-width:544px) {
  .c0 {
    display: none;
  }
}

@media screen and (min-width:768px) {
  .c0 {
    display: block;
  }
}

<div
  className="c0"
  display={
    [
      "none",
      "none",
      "block",
    ]
  }
/>
`;
